### PR TITLE
Validate descriptor component type

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

**Comment by chrisforbes** [comment]
Validates that isampler* is correctly matched with SINT formats, etc.

**Comment by mark-lunarg** [comment]
@tobine would want a test...

**Comment by chrisforbes** [comment]
Adding a test and rebasing.

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -19483,6 +19483,65 @@ TEST_F(VkLayerTest, DrawTimeImageMultisampleMismatchWithPipeline) {
...
+        "}\n";
+    VkShaderObj vs(m_device, vsSource, VK_SHADER_STAGE_VERTEX_BIT, this);
+    VkShaderObj fs(m_device, fsSource, VK_SHADER_STAGE_FRAGMENT_BIT, this);
+
+    VkPipelineObj pipe(m_device);
+    pipe.AddShader(&vs);
+    pipe.AddShader(&fs);
+    pipe.AddDefaultColorAttachment();
+
+    VkTextureObj texture(m_device, nullptr);
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: By default the texture is in a UNORM format, which (if I'm understanding this correctly) why the Draw validation will report the mismatch, yes?  If so, some commentary might be handy as so much of this is hidden behind utility objects.  If not some commentary might be handy to educate the maintain about the "theory of operation"

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -19483,6 +19483,65 @@ TEST_F(VkLayerTest, DrawTimeImageMultisampleMismatchWithPipeline) {
     m_commandBuffer->end();
 }
 
+TEST_F(VkLayerTest, DrawTimeImageComponentTypeMismatchWithPipeline) {
+    TEST_DESCRIPTION(
+        "Test that an error is produced when the component type of an imageview disagrees with the type in the shader.");
+
+    m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, "SINT component type, but bound descriptor");
```
[the remark this answers is not in the document]
Comment (chrisforbes) [reply tests/layer_validation_tests.cpp]: The way DescriptorSet::ValidateDrawState is written forces everything to use the same VUID tag. We should probably rework this function at some point to support case-specific VUIDs.

File: tests/layer_validation_tests.cpp (hunk 1 above)
Comment (chrisforbes) [reply tests/layer_validation_tests.cpp]: Added comment.

**Pull request merged**